<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>E Gynecological and breast cancers data | Supplementary Materials for Probabilistic Graphical Modeling under Heterogeneity</title>
  <meta name="description" content="This containes all the supplementary materials for the paper named Probabilistic Graphical Modeling under Heterogeneity." />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="E Gynecological and breast cancers data | Supplementary Materials for Probabilistic Graphical Modeling under Heterogeneity" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This containes all the supplementary materials for the paper named Probabilistic Graphical Modeling under Heterogeneity." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="E Gynecological and breast cancers data | Supplementary Materials for Probabilistic Graphical Modeling under Heterogeneity" />
  
  <meta name="twitter:description" content="This containes all the supplementary materials for the paper named Probabilistic Graphical Modeling under Heterogeneity." />
  

<meta name="author" content="Liying Chen^{1,5}, Satwik Acharyya^{1,5}, Chunyu Luo^{2,3}, Yang Ni^4 and Veerabhadran Baladandayuthapani^{1,6}" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="StemnessBC.html"/>
<link rel="next" href="ST.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Probabilistic Graphical Model under Heterogeneity</a></li>

<li class="divider"></li>
<li class="appendix"><span><b>Supplementary Materials</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="A" data-path="method.html"><a href="method.html"><i class="fa fa-check"></i><b>A</b> Methodology</a>
<ul>
<li class="chapter" data-level="A.1" data-path="method.html"><a href="method.html#GraphRmodel"><i class="fa fa-check"></i><b>A.1</b> Model and priors</a></li>
<li class="chapter" data-level="A.2" data-path="method.html"><a href="method.html#MFVB"><i class="fa fa-check"></i><b>A.2</b> Mean field variational Bayes</a></li>
<li class="chapter" data-level="A.3" data-path="method.html"><a href="method.html#derivation"><i class="fa fa-check"></i><b>A.3</b> Update equations</a></li>
<li class="chapter" data-level="A.4" data-path="method.html"><a href="method.html#GraphElbo"><i class="fa fa-check"></i><b>A.4</b> Evidence lower bound (ELBO)</a></li>
<li class="chapter" data-level="A.5" data-path="method.html"><a href="method.html#GraphRcompare"><i class="fa fa-check"></i><b>A.5</b> Overview of competing methods</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="simulation.html"><a href="simulation.html"><i class="fa fa-check"></i><b>B</b> Simulation studies</a>
<ul>
<li class="chapter" data-level="B.1" data-path="simulation.html"><a href="simulation.html#undir"><i class="fa fa-check"></i><b>B.1</b> Undirected graphical models</a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="simulation.html"><a href="simulation.html#simhomo"><i class="fa fa-check"></i><b>B.1.1</b> Homogeneous cases</a></li>
<li class="chapter" data-level="B.1.2" data-path="simulation.html"><a href="simulation.html#simgroupspec"><i class="fa fa-check"></i><b>B.1.2</b> Group specific cases</a></li>
<li class="chapter" data-level="B.1.3" data-path="simulation.html"><a href="simulation.html#simcont"><i class="fa fa-check"></i><b>B.1.3</b> Individual-specific cases</a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="simulation.html"><a href="simulation.html#dir"><i class="fa fa-check"></i><b>B.2</b> Directed acyclic graphs (DAGs)</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="simulation.html"><a href="simulation.html#moderate"><i class="fa fa-check"></i><b>B.2.1</b> Moderate dimensions</a></li>
<li class="chapter" data-level="B.2.2" data-path="simulation.html"><a href="simulation.html#largep"><i class="fa fa-check"></i><b>B.2.2</b> Large number of nodes</a></li>
<li class="chapter" data-level="B.2.3" data-path="simulation.html"><a href="simulation.html#largeq"><i class="fa fa-check"></i><b>B.2.3</b> Large number of external covariates</a></li>
</ul></li>
<li class="chapter" data-level="B.3" data-path="simulation.html"><a href="simulation.html#simpson"><i class="fa fa-check"></i><b>B.3</b> Illustration of Simpson’s paradox</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="PAM50.html"><a href="PAM50.html"><i class="fa fa-check"></i><b>C</b> PAM50 proteomics dataset</a>
<ul>
<li class="chapter" data-level="C.1" data-path="PAM50.html"><a href="PAM50.html#PAM50data"><i class="fa fa-check"></i><b>C.1</b> Data description</a></li>
<li class="chapter" data-level="C.2" data-path="PAM50.html"><a href="PAM50.html#PAM50process"><i class="fa fa-check"></i><b>C.2</b> Preprocessing and application</a></li>
<li class="chapter" data-level="C.3" data-path="PAM50.html"><a href="PAM50.html#PAM50result"><i class="fa fa-check"></i><b>C.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="StemnessBC.html"><a href="StemnessBC.html"><i class="fa fa-check"></i><b>D</b> Stemness-induced proteomics dataset</a>
<ul>
<li class="chapter" data-level="D.1" data-path="StemnessBC.html"><a href="StemnessBC.html#StemnessBCdata"><i class="fa fa-check"></i><b>D.1</b> Data description</a></li>
<li class="chapter" data-level="D.2" data-path="StemnessBC.html"><a href="StemnessBC.html#StemnessBCprocess"><i class="fa fa-check"></i><b>D.2</b> Preprocessing and application</a></li>
<li class="chapter" data-level="D.3" data-path="StemnessBC.html"><a href="StemnessBC.html#StemnessBCresult"><i class="fa fa-check"></i><b>D.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="E" data-path="Gyne.html"><a href="Gyne.html"><i class="fa fa-check"></i><b>E</b> Gynecological and breast cancers data</a>
<ul>
<li class="chapter" data-level="E.1" data-path="Gyne.html"><a href="Gyne.html#Gynedata"><i class="fa fa-check"></i><b>E.1</b> Data description</a></li>
<li class="chapter" data-level="E.2" data-path="Gyne.html"><a href="Gyne.html#Gyneprocess"><i class="fa fa-check"></i><b>E.2</b> Preprocessing and application</a></li>
<li class="chapter" data-level="E.3" data-path="Gyne.html"><a href="Gyne.html#Gyneresult"><i class="fa fa-check"></i><b>E.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="F" data-path="ST.html"><a href="ST.html"><i class="fa fa-check"></i><b>F</b> Spatial transcriptomics dataset</a>
<ul>
<li class="chapter" data-level="F.1" data-path="ST.html"><a href="ST.html#STdata"><i class="fa fa-check"></i><b>F.1</b> Data description</a></li>
<li class="chapter" data-level="F.2" data-path="ST.html"><a href="ST.html#STprocess"><i class="fa fa-check"></i><b>F.2</b> Preprocessing and application</a></li>
<li class="chapter" data-level="F.3" data-path="ST.html"><a href="ST.html#STBCresult"><i class="fa fa-check"></i><b>F.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="G" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html"><i class="fa fa-check"></i><b>G</b> Implementation</a>
<ul>
<li class="chapter" data-level="G.1" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#GraphRpackage"><i class="fa fa-check"></i><b>G.1</b> GraphR package</a>
<ul>
<li class="chapter" data-level="G.1.1" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#installation"><i class="fa fa-check"></i><b>G.1.1</b> Installation</a></li>
<li class="chapter" data-level="G.1.2" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#graphr_est-function"><i class="fa fa-check"></i><b>G.1.2</b> GraphR_est() function</a></li>
<li class="chapter" data-level="G.1.3" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#graphr_pred-function"><i class="fa fa-check"></i><b>G.1.3</b> GraphR_pred() function</a></li>
<li class="chapter" data-level="G.1.4" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#graphr_visualization-function"><i class="fa fa-check"></i><b>G.1.4</b> GraphR_visualization() function</a></li>
</ul></li>
<li class="chapter" data-level="G.2" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#example"><i class="fa fa-check"></i><b>G.2</b> Example</a>
<ul>
<li class="chapter" data-level="G.2.1" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#example-1"><i class="fa fa-check"></i><b>G.2.1</b> Example</a></li>
</ul></li>
<li class="chapter" data-level="G.3" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#GraphRshinyApp"><i class="fa fa-check"></i><b>G.3</b> GraphR Shiny App and tutorial website</a></li>
<li class="chapter" data-level="G.4" data-path="ImplementGraphR.html"><a href="ImplementGraphR.html#checkmarks"><i class="fa fa-check"></i><b>G.4</b> Checkmarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Supplementary Materials for Probabilistic Graphical Modeling under Heterogeneity</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Gyne" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">E</span> Gynecological and breast cancers data<a href="Gyne.html#Gyne" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="Gynedata" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">E.1</span> Data description<a href="Gyne.html#Gynedata" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Gynecological and breast cancer (Pan-gyane) proteomics dataset was obtained from The Cancer Genome Atlas (TCGA, <span class="citation">Weinstein et al. (<a href="#ref-weinstein2013cancer" role="doc-biblioref">2013</a>)</span>) and The Cancer Proteome Atlas (TCPA, <span class="citation">Li et al. (<a href="#ref-li2013tcpa" role="doc-biblioref">2013</a>)</span>). The Pan-gyane proteomics contains four types of cancers including breast invasive carcinoma (BRCA), cervical squamous cell carcinoma and endocervical adenocarcinoma (CESC), ovarian serous cystadenocarcinoma (OV) and uterine corpus endometrial carcinoma (UCEC). Abundance of 189 kinds of protein were measured across 1,941 patients from <span class="citation">Weinstein et al. (<a href="#ref-weinstein2013cancer" role="doc-biblioref">2013</a>)</span> among which 892 patients had BRCA, 173 had CESC, 436 had OV and 440 had UCEC.</p>
</div>
<div id="Gyneprocess" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">E.2</span> Preprocessing and application<a href="Gyne.html#Gyneprocess" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>All settings are same as in Section <a href="PAM50.html#PAM50process">C.2</a>, except that:</p>
<ol style="list-style-type: upper-roman">
<li><p>Four external covariates: indicators of BRCA, CESC, OV, UCEC are included in the analyses.</p></li>
<li><p>Significantly strong correlation is defined as connection with FDR based p-values <span class="math inline">\(&lt;0.01\)</span> and magnitude <span class="math inline">\(\mid \rho \mid \geq 0.4\)</span>)</p></li>
</ol>
</div>
<div id="Gyneresult" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">E.3</span> Results<a href="Gyne.html#Gyneresult" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Figure <a href="Gyne.html#fig:panpip">E.1</a> is a heatmap which shows posterior inclusion probability (PIP) of selected edges in each type of cancer.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:panpip"></span>
<img src="images/pan_pip.jpeg" alt="Posterior inclusion probability (PIP) of selected edges in each type of cancer." width="90%" />
<p class="caption">
Figure E.1: Posterior inclusion probability (PIP) of selected edges in each type of cancer.
</p>
</div>
<p>Figure <a href="Gyne.html#fig:brcafullnet">E.2</a>, <a href="Gyne.html#fig:cescfullnet">E.3</a>, <a href="Gyne.html#fig:ovfullnet">E.4</a> and <a href="Gyne.html#fig:ucecfullnet">E.5</a> show networks of selected protein pairs in BRCA, CESC, OV and UCEC respectively. All other settings are same as figure <a href="PAM50.html#fig:lumfullnet">C.2</a></p>
<p><strong>BRCA</strong></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:brcafullnet"></span>
<img src="images/pan_fullnet_brca.jpeg" alt="Network of selected protein pairs in BRCA patients." width="70%" />
<p class="caption">
Figure E.2: Network of selected protein pairs in BRCA patients.
</p>
</div>
<p><strong>CSEC</strong></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:cescfullnet"></span>
<img src="images/pan_fullnet_cesc.jpeg" alt="Network of selected protein pairs in CESC patients." width="70%" />
<p class="caption">
Figure E.3: Network of selected protein pairs in CESC patients.
</p>
</div>
<p><strong>OV</strong></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ovfullnet"></span>
<img src="images/pan_fullnet_ov.jpeg" alt="Network of selected protein pairs in OV patients." width="70%" />
<p class="caption">
Figure E.4: Network of selected protein pairs in OV patients.
</p>
</div>
<p><strong>UCEC</strong></p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ucecfullnet"></span>
<img src="images/pan_fullnet_ucec.jpeg" alt="Network of selected protein pairs in UCEC patients." width="70%" />
<p class="caption">
Figure E.5: Network of selected protein pairs in UCEC patients.
</p>
</div>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-li2013tcpa" class="csl-entry">
Li, Jun, Yiling Lu, Rehan Akbani, Zhenlin Ju, Paul L Roebuck, Wenbin Liu, Ji-Yeon Yang, et al. 2013. <span>“TCPA: A Resource for Cancer Functional Proteomics Data.”</span> <em>Nature Methods</em> 10 (11): 1046–47.
</div>
<div id="ref-weinstein2013cancer" class="csl-entry">
Weinstein, John N, Eric A Collisson, Gordon B Mills, Kenna R Shaw, Brad A Ozenberger, Kyle Ellrott, Ilya Shmulevich, Chris Sander, and Joshua M Stuart. 2013. <span>“The Cancer Genome Atlas Pan-Cancer Analysis Project.”</span> <em>Nature Genetics</em> 45 (10): 1113–20.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="StemnessBC.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="ST.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-Gyne-BC.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
